code/modules/fish/fish_tank.dm

```diff
@@ -0,0 +1,716 @@
+			if(MINOR_LEAK)
+				examine_message += "<span class = 'warning'>\The [src] is leaking.</span>"
+			if(MAJOR_LEAK)
+				examine_message += "<span class = 'warning'>\The [src] is leaking profusely!</span>"
+	else
+		switch (leaking)									//No water, report the cracks instead
+			if(MINOR_LEAK)
+				examine_message += "<span class = 'warning'>\The [src] is cracked.</span>"
+			if(MAJOR_LEAK)
+				examine_message += "<span class = 'warning'>\The [src] is nearly shattered!</span>"
+
+
+	//Finally, report the full examine_message constructed from the above reports
+	to_chat(user, "[jointext(examine_message, "")]")
+	return examine_message
+
+//////////////////////////////
+//		ATACK PROCS			//
+//////////////////////////////
+
+/obj/machinery/fishtank/attack_alien(mob/living/user as mob)
+	if(islarva(user))
+		return
+	attack_generic(user, 15)
+
+
+/obj/machinery/fishtank/attack_hand(mob/user as mob)
+
+	if(user.a_intent == I_HURT)
+		playsound(get_turf(src), 'sound/effects/glassknock.ogg', 80, 1)
+		user.visible_message("<span class='danger'>\The [user] bangs against the [src]!</span>", \
+							"<span class='danger'>You bang against the [src]!</span>", \
+							"You hear a banging sound.")
+	else
+		playsound(src.loc, 'sound/effects/glassknock.ogg', 80, 1)
+		user.visible_message("\The [user] taps on \the [src].", \
+							"You tap on \the [src].", \
+							"You hear a knocking sound.")
+
+
+/obj/machinery/fishtank/proc/hit(var/damage, var/mob/user, var/obj/O)
+	user.delayNextAttack(0.3 SECONDS)
+	user.do_attack_animation(src, O) // Ensuring the legacy lives on :^)
+	playsound(get_turf(src), 'sound/effects/glassknock.ogg', 80, 1)
+	cur_health = max(0, cur_health - damage)
+	check_health()
+
+/obj/machinery/fishtank/proc/attack_generic(mob/living/user as mob, damage = 0)	//used by attack_alien, attack_animal, and attack_slime
+	cur_health = max(0, cur_health - damage)
+	if(cur_health <= 0)
+		user.visible_message("<span class='danger'>\The [user] smashes through \the [src]!</span>")
+		destroy()
+	else	//for nicer text
+		user.visible_message("<span class='danger'>\The [user] smashes into \the [src]!</span>")
+		playsound(loc, 'sound/effects/Glasshit.ogg', 100, 1)
+		check_health()
+
+/obj/machinery/fishtank/attackby(var/obj/item/O, var/mob/user as mob)
+	//Welders repair damaged tanks on help intent, damage on all others
+	if(issilicatesprayer(O))
+		var/obj/item/device/silicate_sprayer/S = O
+		if(user.a_intent == I_HELP)
+			if (S.get_amount() > 2)
+				if(cur_health < max_health)
+					to_chat(user, "<span class='notice'>You repair some of the cracks on \the [src].</span>")
+					cur_health += 20
+					S.remove_silicate(2)
+					check_health()
+				else
+					to_chat(user, "<span class='warning'>There is no damage to fix!</span>")
+			else if (cur_health < max_health)
+				to_chat(user, "<span class='notice'>You require more silicate to fix the damage on \the [src].</span>")
+		else
+			hit(S.force, user, O)
```

Comment:
silicate sprayer has `force = 0` anyway, I assume this is a left-over from when it used welders